crates/runix/src/flake_ref/mod.rs

```diff
@@ -213,16 +214,25 @@ impl FlakeRef {
     }
 
     /// Parses a URI into a flake reference given the URI and the path to the `parser-util` binary
-    pub fn from_url<T, P>(uri: T, bin_path: P) -> Result<Self, UrlParseError>
+    pub fn from_url<U, P>(uri: U, bin_path: P) -> Result<Self, UrlParseError>
```

Comment:
Why not just

```suggestion
    pub fn from_url<U, P>(url_str: impl AsRef<str>, bin_path: impl AsRef<Path>) -> Result<Self, UrlParseError>
```

and then call `url_parser::installable_flake_ref(url_str, bin_path)?` below? I dont think we need to add a `Clone` bound..

Reply:
The `Clone` bound is a mistake. It's a leftover from when I was cloning the URL to store on the `IndirectRef`.